cpp/test/optionsTest.cc

```diff
@@ -0,0 +1,299 @@
+  setAttributeOptions->setRecursive(true);
+  bool recursive = setAttributeOptions->isRecursive();
+  assert(recursive == true);
+  Status status1 = setAttributeOptions->setOwner("user1");
+  Status status2 = setAttributeOptions->setGroup("group1");
+  assert(status1.ok());
+  assert(status2.ok());
+  std::string owner = setAttributeOptions->getOwner();
+  std::string group = setAttributeOptions->getGroup();
+  assert(owner.compare("user1") == 0);
+  assert(group.compare("group1") == 0);
+}
+
+void GetStatusOptionsTest() {
+  GetStatusOptions* getStatusOptions = GetStatusOptions::getDefaultOptions();
+  getStatusOptions->setLoadMetadataType(new LoadMetadataType(2));
+  LoadMetadataType* loadMetadataType = getStatusOptions->getLoadMetadataType();
+  assert(loadMetadataType->getValue() == 2);
+}
+
+void getWorkerOptionsTest() {
+  WorkerNetAddress* netAddress = new WorkerNetAddress("host1", 1, 1, 1, "1");
+  WorkerNetAddress* netAddress2 = new WorkerNetAddress(
+      "192.168.195.132", 2, 2, 2, "2");
+  BlockWorkerInfo* workerInfo = new BlockWorkerInfo(netAddress, 20, 2);
+  BlockWorkerInfo* workerInfo2 = new BlockWorkerInfo(netAddress2, 10, 2);
+  std::vector<BlockWorkerInfo> workerInfoList;
+  workerInfoList.push_back(*workerInfo);
+  workerInfoList.push_back(*workerInfo2);
+  GetWorkerOptions* getWorkerOptions = GetWorkerOptions::getDefaultOptions();
+  getWorkerOptions->setBlockWorkerInfos(&workerInfoList);
+  std::vector<BlockWorkerInfo>* newInfoList =
+      getWorkerOptions->getBlockWorkerInfos();
+  assert(newInfoList->begin()->getUsedBytes() == 2);
+  FileWriteLocationPolicy* policy = NULL;
+  policy = LocalFirstAvoidEvictionPolicy::getPolicy();
+  WorkerNetAddress* newAddress = policy->getWorkerForNextBlock(
+      getWorkerOptions->getBlockWorkerInfos(),
+      getWorkerOptions->getBlockSize());
+  assert(newAddress->getHost().compare("192.168.195.132") == 0);
+  BlockLocationPolicy* newPolicy = NULL;
+  newPolicy = DeterministicHashPolicy::getPolicy();
+  WorkerNetAddress* address = newPolicy->getWorker(getWorkerOptions);
+  assert(address->getHost().compare("host1") == 0);
+  newPolicy = SpecificHostPolicy::getPolicy("192.168.195.132");
+  address = newPolicy->getWorker(getWorkerOptions);
+  assert(address->getHost().compare("192.168.195.132") == 0);
+  newPolicy = MostAvailableFirstPolicy::getPolicy();
+  address = newPolicy->getWorker(getWorkerOptions);
+  assert(address->getHost().compare("host1") == 0);
+}
+
+int main(void) {
+  FileSystem* fileSystem;
+  LocalAlluxioCluster* miniCluster = new LocalAlluxioCluster();
+  miniCluster->start();
+  miniCluster->getClient(&fileSystem);
+  CreateDirectoryOptionsTest();
+  CreateFileOptionsTest();
+  DeleteOptionsTest();
+  ExistsOptionsTest();
+  FreeOptionsTest();
+  ListStatusOptionsTest();
+  MountOptionsTest();
+  OpenFileOptionsTest();
+  SetAttributeOptionsTest();
+  GetStatusOptionsTest();
+  getWorkerOptionsTest();
+  delete miniCluster;
```

Comment:
`delete fileSystem`?

Reply:
What you required has all been changed, and the Jenkins error has been fixed.